### PR TITLE
feat(tokens)!: Extract RGB color values into their own custom properties [CSS-383]

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description

To solve our problem of needing to decouple the rgb values from their opacities in order to utilize them in our CSS while also continuing to support the existing values, this PR adds a new postcss plugin called `postcss-rgb-mapping`.

This is a custom written plugin for this purpose but published separately: https://github.com/castastrophe/postcss-rgb-mapping

This plugin turns this:

```css
.spectrum--lightest {
  --spectrum-seafoam-100: rgba(206, 247, 243);
  --spectrum-seafoam-200: rgba(170, 241, 234, 0.5);
}
```

Into this:

```css
.spectrum--lightest {
  --spectrum-seafoam-100-rgb: 206, 247, 243;
  --spectrum-seafoam-100: rgba(var(--spectrum-seafoam-100-rgb));
  --spectrum-seafoam-200-rgb: 170, 241, 234;
  --spectrum-seafoam-200-opacity: 0.5;
  --spectrum-seafoam-200: rgba(var(--spectrum-seafoam-200-rgb), var(--spectrum-seafoam-200-opacity));
}
```

To maintain the existing codebase and add the rgb tokens, this parser separates out the rgb and opacity values and updates the old token to point to these new values. This approach maintains a single-source-of-truth while allowing for backwards compatibility.

## To-do list
- [x] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [x] This pull request is ready to merge.

## References

- [fastmod](https://github.com/facebookincubator/fastmod) A possible option for shipping a tool to downstream users making it easier for them to make this update.

### CODE REVIEW

**Comment by castastrophe** [comment]
@Rajdeepc @pfulton and @GarthDB, I've updated the excellent work Rajdeep did to match the solution you & I had discussed Garth. This version updates all rgb or rgba values to their raw formats (`255, 255, 255` rather than `rgb(255,255,255)`) and shows how the CSS needs to be updated.

File: components/tokens/config.js
Comment (castastrophe) [inline components/tokens/config.js:1]: Moved this file to style-dictionary.config.js to delineate it from postcss.config.js.

**Comment by castastrophe** [comment]
RGB postcss tool: https://github.com/castastrophe/postcss-rgb-mapping

**Comment by castastrophe** [comment]
Notes on some of the scope creep going on in this PR:

- Cleans up tokens & vars build process to use little to no gulp
- Sets up tokens to build separately - moves us towards building them separate from the build - ideally they only get recompiled when the package changes
- Tokens is now using postcss to structure the output, eventually much of this can be pulled out to be configured at a higher level b/c the same rules apply to the other packages too
- Removed a lot of unused code from the vars updateDNA file and the main gulpfile as well
- Fixes the release build task too!

File: components/expressvars/tasks/updateDNA.js [hunk 1]
```diff
@@ -207,7 +158,7 @@ function calculateOverrides(objects, processValue) {
 }
 
 function generateDNAFiles() {
-  const dnaJSONPath = path.join(path.dirname(require.resolve('@adobe/spectrum-express-tokens-deprecated')), '..', 'dist', 'data', 'json', 'dna-linked.json');
+  const dnaJSONPath = path.join(path.dirname(require.resolve('@adobe/spectrum-tokens-deprecated')), '..', 'dist', 'data', 'json', 'dna-linked.json');
```
Comment (castastrophe) [inline components/expressvars/tasks/updateDNA.js:161]:
```suggestion
  const dnaJSONPath = path.join(path.dirname(require.resolve('@adobe/spectrum-express-tokens-deprecated')), '..', 'dist', 'data', 'json', 'dna-linked.json');
```

File: components/expressvars/package.json [hunk 2]
```diff
@@ -15,17 +15,21 @@
...
-    "update": "gulp update"
+    "build": "yarn clean && gulp build",
+    "clean": "rimraf dist/*",
+    "copy:metadata": "cp json/spectrum-metadata.json dist/",
+    "dna:clean": "rimraf css/* js/*",
+    "dna:update": "yarn dna:clean && gulp update && yarn build"
   },
   "devDependencies": {
-    "del": "^5.0.0",
+    "@adobe/spectrum-tokens-deprecated": "^11.8.0",
```
Comment (castastrophe) [inline components/expressvars/package.json:25]:
```suggestion
    "@adobe/spectrum-express-tokens-deprecated": "^11.8.0",
```

**Review by castastrophe — APPROVED** [review]

**Review by pfulton — APPROVED** [review]

**Review by pfulton — APPROVED** [review]

**Review by pfulton — APPROVED** [review]

**Pull request merged**